apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -1274,12 +1295,22 @@ void scheduleScan(int height, boolean validate) {
 
     @Override
     public void scan(int height, boolean validate) {
-        scan(height, validate, false);
+        try {
+            trimService.updateTrimConfig(false, true);
```

Comment:
Do not disable trim during scan. TrimObserver.onBlockScanned will not be triggered.

Reply:
Sure, TrimObserver.onBlockScanned is not triggered, but unfortunately, the TrimService periodical thread is the main issue of deadlock. So, I disable that thread by calling the trimService.updateTrimConfig(false, true);

Reply:
done